File: browser-extension/scripts/generate_icons.mjs
#!/usr/bin/env node
/**
 * Generate placeholder icons for the Brain Web extension.
 * Creates simple colored icons with a "BW" text overlay.
 */

import { createCanvas } from 'canvas';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function createIcon(size, outputPath) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Background gradient
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#4A90E2');
  gradient.addColorStop(1, '#2C5F8D');
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, size, size);
  
  // Rounded rectangle overlay
  const margin = size / 8;
  const radius = size / 6;
  ctx.fillStyle = '#2C5F8D';
  ctx.strokeStyle = '#1A3A5C';
  ctx.lineWidth = Math.max(1, size / 32);
  
  ctx.beginPath();
  ctx.roundRect(margin, margin, size - 2 * margin, size - 2 * margin, radius);
  ctx.fill();
  ctx.stroke();
  
  // Add "BW" text
  ctx.fillStyle = '#FFFFFF';
  ctx.font = `bold ${Math.floor(size / 2)}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('BW', size / 2, size / 2);
  
  // Save as PNG
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(outputPath, buffer);
  console.log(`Created ${outputPath} (${size}x${size})`);
}

function main() {
  const assetsDir = path.join(__dirname, '..', 'assets');
  if (!fs.existsSync(assetsDir)) {
    fs.mkdirSync(assetsDir, { recursive: true });
  }
  
  const sizes = [16, 48, 128];
  for (const size of sizes) {
    const outputPath = path.join(assetsDir, `icon${size}.png`);
    createIcon(size, outputPath);
  }
  
  console.log(`\nIcons created in ${assetsDir}`);
}

main();

File: browser-extension/scripts/generate_icons.py
#!/usr/bin/env python3
"""
Generate placeholder icons for the Brain Web extension.
Creates simple colored icons with a "BW" text overlay.
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, output_path):
    """Create a simple icon with Brain Web branding."""
    # Create image with a gradient-like background
    img = Image.new('RGB', (size, size), color='#4A90E2')
    draw = ImageDraw.Draw(img)
    
    # Draw a simple circle/rounded square background
    margin = size // 8
    draw.rounded_rectangle(
        [margin, margin, size - margin, size - margin],
        radius=size // 6,
        fill='#2C5F8D',
        outline='#1A3A5C',
        width=max(1, size // 32)
    )
    
    # Add "BW" text
    try:
        # Try to use a system font, fallback to default
        font_size = size // 2
        try:
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
        except:
            try:
                font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", font_size)
            except:
                font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()
    
    text = "BW"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    position = ((size - text_width) // 2, (size - text_height) // 2 - bbox[1])
    draw.text(position, text, fill='#FFFFFF', font=font)
    
    img.save(output_path, 'PNG')
    print(f"Created {output_path} ({size}x{size})")

def main():
    assets_dir = os.path.join(os.path.dirname(__file__), '..', 'assets')
    os.makedirs(assets_dir, exist_ok=True)
    
    sizes = [16, 48, 128]
    for size in sizes:
        output_path = os.path.join(assets_dir, f'icon{size}.png')
        create_icon(size, output_path)
    
    print(f"\nIcons created in {assets_dir}")

if __name__ == '__main__':
    main()

